scipy/spatial/transform/rotation.py

```diff
@@ -1460,3 +1461,109 @@ def __getitem__(self, indexer):
                [ 0.57735027,  0.57735027, -0.57735027,  0.        ]])
         """
         return self.__class__(self._quat[indexer], normalized=True)
+
+    @classmethod
+    def match_vectors(cls, a, b, weights=None, normalized=False):
+        """Estimate the rotation to match two sets of vectors.
+
+        Find a rotation between frames A and B which best matches a set of unit
+        vectors `a` and `b` observed in these frames. The following loss
+        function is minimized to solve for the direction cosine matrix
+        :math:`C`:
+
+        .. math::
+
+            L(C) = \\frac{1}{2} \\sum_{i = 1}^{n} w_i \\lVert \\mathbf{a}_i -
+            C \\mathbf{b}_i \\rVert^2 ,
+
+        where :math:`w_i`'s are the `weights` corresponding to each vector.
+
+        The rotation is estimated using Markley's SVD method [1]_.
+
+        Parameters
+        ----------
+        a : array_like, shape (N, 3)
+            Vectors expressed in initial frame A. Result of applying an
+            arbitrary rotation to `b`. Each `a[i]` corresponds to a vector.
+        b : array_like, shape (N, 3)
+            Vectors expressed in another frame B. Shape must match `a`. Each
+            `b[i]` corresponds to a vector.
+        weights : None or array_like, 1D
+            Coefficients describing the relative importance of the vectors in
+            `a`. Number of values in `weights` must match number of vectors
+            specified in `a` and `b`. `weights` are normalized before matching
+            vectors. If None, then all values in `weights` are assumed to be
+            equal. Default is None.
+        normalized : boolean, optional
+            If True, assume input vectors `a` and `b` to have unit norm. If
+            False, normalize `a` and `b` before estimating rotation. Default
+            is False.
+
+        Returns
+        -------
+        estimated_rotation : `Rotation` instance
+            Best estimate of the rotation that transforms `b` to `a`.
+        covariance_matrix : `numpy.ndarray`, shape (3, 3)
+            Covariance matrix of the three component error vector of the Euler
+            angles describing the `estimated_rotation`.
+
+        References
+        ----------
+        .. [1] F. Landis Markley,
+                “Attitude determination using vector observations: a fast
+                optimal matrix algorithm,” Journal of Astronautical Sciences,
+                Vol. 41, No.2, 1993, pp. 261-280.
+        """
+        outvecs = np.asarray(a)
```

Comment:
Maybe keep names as `a` and `b`?

Reply:
I see what you mean. In this case maybe we should use similar names in the docstring as well?

@ilayn what is your opinion?